File: FireflowSuite/engine/src/org/fireflow/engine/definition/IDefinitionService.java
/**
 * Copyright 2007-2008 陈乜云（非也,Chen Nieyun）
 * All rights reserved. 
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation。
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses. *
 */
package org.fireflow.engine.definition;

import java.util.List;
import org.fireflow.engine.IRuntimeContextAware;

/**
 * @author chennieyun
 *
 */
public interface IDefinitionService extends IRuntimeContextAware {
//	public WorkflowProcess getWorkflowProcessByName(String name);
//	public void setDefinitionFiles(List<String> definitionFileNames)throws IOException,FPDLParserException;
//	public List<WorkflowProcess> getAllWorkflowProcesses();
//        public WorkflowProcess getWorkflowProcessById(String id);
    
    public List<WorkflowDefinition> getAllLatestVersionOfWorkflowDefinitions();
    public WorkflowDefinition getWorkflowDefinitionByProcessIdAndVersion(String processId ,Integer version);
    public WorkflowDefinition getTheLatestVersionOfWorkflowDefinition(String processId);
}

File: trunk/dot-net_version/FireWorkflow.Net/FireWorkflow.Net/Engine/Definition/IDefinitionService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FireWorkflow.Net.Engine.Definition
{

    /// <summary>流程定义服务。</summary>
    public interface IDefinitionService : IRuntimeContextAware
    {
        //	public WorkflowProcess getWorkflowProcessByName(String name);
        //	public void setDefinitionFiles(List<String> definitionFileNames)throws IOException,FPDLParserException;
        //	public List<WorkflowProcess> getAllWorkflowProcesses();
        //        public WorkflowProcess getWorkflowProcessById(String id);

        /// <summary>返回所有流程的最新版本</summary>
        /// <returns></returns>
         List<WorkflowDefinition> getAllLatestVersionsOfWorkflowDefinition();


        /// <summary>根据流程Id和版本号查找流程定义</summary>
        WorkflowDefinition getWorkflowDefinitionByProcessIdAndVersionNumber(String processId, Int32? version);

        /// <summary>通过流程Id查找其最新版本的流程定义</summary>
        WorkflowDefinition getTheLatestVersionOfWorkflowDefinition(String processId);
    }
}
